File: src/csharp/Ticketing.Domain/Ticket.cs
namespace Ticketing.Domain;

public class Ticket
{
    public string Id { get; }
    public string Title { get; private set; }
    public string Description { get; private set; }
    public TicketStatus Status { get; private set; }
    public string CreatedByUserId { get; }
    public string? AssigneeUserId { get; private set; }

    public Ticket(
        string id,
        string title,
        string description,
        string createdByUserId)
    {
        Id = id;
        Title = title;
        Description = description;
        CreatedByUserId = createdByUserId;
        Status = TicketStatus.Open;
        AssigneeUserId = null;
    }

    public void AssignTo(string userId)
    {
        AssigneeUserId = userId;
    }
}

File: src/kotlin/ticketing-domain/src/main/kotlin/ticketing/domain/Ticket.kt
package ticketing.domain

class Ticket(
    val id: String,
    title: String,
    description: String,
    val createdByUserId: String
) {
    var title: String = title
        private set

    var description: String = description
        private set

    var status: TicketStatus = TicketStatus.Open
        private set

    var assigneeUserId: String? = null
        private set

    init {
        // TODO:
        // Reject blank id.
        // Reject blank title.
        // Reject blank createdByUserId.
        //
        // Kotlin equivalent:
        // if (id.isBlank()) { throw IllegalArgumentException("...") }
    }

    fun assignTo(userId: String) {
        // TODO:
        // Reject blank userId.
        // Then assign assigneeUserId.
    }

    fun changeStatus(newStatus: TicketStatus) {
        // TODO:
        // Implement same allowed transitions as C#:
        // Open -> InProgress
        // InProgress -> Resolved
        // Resolved -> Closed
        // Everything else throws IllegalStateException.
    }

    private fun formatStatus(status: TicketStatus): String {
        // TODO:
        // Return:
        // Open -> "Open"
        // InProgress -> "In Progress"
        // Resolved -> "Resolved"
        // Closed -> "Closed"

        return ""
    }
}